### ISSUE TITLE
New rule: string-quotes

### ISSUE THREAD

**Issue by davidtheclark** [stylelint/stylelint#96]
I'm getting confused by scanning back through the comments about strings. Do either of you have a strong sense for where this discussion concluded? Is `string-quotes` a catch-all rule for quotes anywhere (attributes, content, functions like url, etc.)?

**Comment by MoOx** [comment]
> Is string-quotes a catch-all rule for quotes anywhere (attributes, content, functions like url, etc.)? 

I think it should.

**Comment by jeddy3** [comment]
> Do either of you have a strong sense for where this discussion concluded?

No, not a strong sense. However, I remember thinking it was *not* a catch-all for *quotes everywhere*, but rather a catch-all for everything *not* covered by `url-quotes` and `selector-attribute-quotes` i.e. mainly `content: ...`.

Am I right in thinking that if this rule does act as a catch-all for *quotes everywhere* then a user can *not* enforce a styleguide that specifies no quotes for urls, single quotes for strings and double quotes for attribute selectors as the `string-quotes: "single"` will flag warnings when it comes across double quotes for attribute selectors and no quotes for urls?

**Comment by davidtheclark** [comment]
You're right about the implications of a catch-all for quotes.

Right now the rule will ignore `url()` functions that *don't* have quotes. It will only correct those that use a single quote where a double quote is expected, or the other way around. (We would have a contradiction if some user set "double" for this and then "single" for the url-quotes rule.)

I think it's very reasonable to assume that users will want a consistent quote style wherever there are quotes (if they want no quotes for urls, that's a different story). I am not sure that distinguishing selector attribute quotes from all other "string" quotes makes a ton of sense *unless we get even more granular*: if the principle is that we're going to allow the user to specify a quote style in every location in which a string can be used, then it would be consistent; if not, isn't it a kind of arbitrary split to single out selector attributes?

**Comment by MoOx** [comment]
I would go for a simple unique rule for now since this will cover most use cases, probably for 99% of the users.

**Comment by jeddy3** [comment]
> isn't it a kind of arbitrary split to single out selector attributes?

Good point.

> Maybe we could balance granularity and say "selector-quotes" vs. "value-quotes" ... but is that the whole picture?

Also, a good point. 

> I think it's very reasonable to assume that users will want a consistent quote style wherever there are quotes (if they want no quotes for urls, that's a different story)

Agreed. Lets keep this rule as it is. Like @MoOx said, it'll probably cover 99% of the use cases :)

**Comment by davidtheclark** [comment]
Ok, so we stick with this as-is for now and eliminate `selector-attribute-quotes` from #1?

**Comment by MoOx** [comment]
lgtm

**Comment by corysimmons** [comment]
voting for a difference in selector quotes, value quotes, and function quotes.

```css
[class*="foo"] {
  content: '';
  background: url(foo.jpg);
}
```

is best css.
